plot_bdb.py

```diff
@@ -0,0 +1,234 @@
+
+
+def __parse_args():
+  parser = argparse.ArgumentParser(
+    description="Generate graphs for Big Data Benchmark experiments.")
+  parser.add_argument(
+    "-r",
+    "--results-dir",
+    help=("The location of the experiment results. See the comment at the top of this script " +
+      "for details on how the results directory should be structured."),
+    required=True)
+  parser.add_argument(
+    "-o",
+    "--output-dir",
+    help="The directory in which to store the graph and any supporting files.",
+    required=True)
+  parser.add_argument(
+    "-m",
+    "--monotasks-branch",
+    help=("The branch of the 'spark-monotasks' repository that was used to execute the Monotasks " +
+      "trials."),
+    required=True)
+  parser.add_argument(
+    "-s",
+    "--spark-branch",
+    help=("The branch of the 'spark-monotasks' repository that was used to execute the Spark " +
+      "trials."),
+    required=True)
+  parser.add_argument(
+    "-w",
+    "--num_warmup-trials",
+    help="The number of trials to treat as warmup runs and discard.",
+    required=True,
+    type=int)
+  parser.add_argument(
+    "-c",
+    "--plot-continuous-monitors",
+    action="store_true",
+    default=False,
+    help=("If present, will plot all continuous monitors, whose names must end with " +
+      "'executor_monitor'"),
+    required=False)
+  return parser.parse_args()
+
+
+def __find_event_logs(args):
+  """
+  Navigates the filesystem hierarchy described in the comment at the top of this script and creates
+  the following dictionary:
+    { query name : ( monotasks event log file, spark event log file ) }
+  """
+  def error(log_file, program, query):
+    raise Exception(
+      "Unable to find {} file for branch {} for query {}".format(log_file, program, query))
+
+  results_dir = args.results_dir
+  monotasks_branch = args.monotasks_branch
+  spark_branch = args.spark_branch
+  plot_continuous_monitors = args.plot_continuous_monitors
+  queries = {}
+
+  for query_name in os.listdir(results_dir):
+    # For every query...
+    query_dir = path.join(results_dir, query_name)
+
+    if (path.isdir(query_dir)):
+      monotasks_event_log = None
+      spark_event_log = None
+
+      for branch_name in os.listdir(query_dir):
+        # For every branch...
+        is_monotasks_branch = branch_name == monotasks_branch
+        is_spark_branch = branch_name == spark_branch
+
+        # We only care about these results if they're for Monotasks or Spark.
+        if is_monotasks_branch or is_spark_branch:
+          branch_dir = path.join(query_dir, branch_name)
+          if (path.isdir(branch_dir)):
+            if plot_continuous_monitors:
+              utils.plot_continuous_monitors(branch_dir)
+
+            event_log = path.join(branch_dir, "event_log")
+            if is_monotasks_branch:
+              monotasks_event_log = event_log
+            elif is_spark_branch:
+              spark_event_log = event_log
+            else:
```

Comment:
It doesn't look like this else-statement ever gets evaluated (was it supposed to be moved back 4 spaces)?  I'd eliminate the if-statement on line 124 (so you always get the branch directory) and then leave the rest of the code as-is, since it seems like the case where the input is *not* the Spark or monotasks branch should be rare.